lib/jsdom/browser/Window.js

```diff
@@ -330,55 +327,151 @@ function Window(options) {
 
   ///// METHODS
 
+  // https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#timers
+
+  // In the spec the list of active timers is a set of IDs. We make it a map of IDs to Node.js timer objects, so that
+  // we can call Node.js-side clearTimeout() when clearing, and thus allow process shutdown faster.
+  const listOfActiveTimers = new Map();
   let latestTimerId = 0;
-  let latestAnimationFrameCallbackId = 0;
 
-  this.setTimeout = function (fn, ms) {
-    const args = [];
-    for (let i = 2; i < arguments.length; ++i) {
-      args[i - 2] = arguments[i];
+  this.setTimeout = function (handler, timeout = 0, ...args) {
+    if (typeof handler !== "function") {
+      handler = webIDLConversions.DOMString(handler);
+    }
+    timeout = webIDLConversions.long(timeout);
+
+    return timerInitializationSteps(handler, timeout, args, { methodContext: window, repeat: false });
+  };
+  this.setInterval = function (handler, timeout = 0, ...args) {
+    if (typeof handler !== "function") {
+      handler = webIDLConversions.DOMString(handler);
+    }
+    timeout = webIDLConversions.long(timeout);
+
+    return timerInitializationSteps(handler, timeout, args, { methodContext: window, repeat: true });
+  };
+
+  this.clearTimeout = function (handle) {
```

Comment:
https://html.spec.whatwg.org/multipage/webappapis.html#windoworworkerglobalscope has the following IDL:

```webidl
void clearTimeout(optional long handle = 0);
```

We should call `webIDLConversions.long` on and provide the default value for `handle`.

Reply:
I also realized the same thing applies to `clearInterval`.

Reply:
This doesn't seem to have been fixed.